cherrypy/lib/auth_digest.py

```diff
@@ -109,20 +150,23 @@ def get_ha1_file_htdigest(filename):
     argument be an absolute path, to avoid problems.
     """
 
-    def get_ha1(realm, username):
+    def get_ha1(**kwargs):
         result = None
-        with open(filename, 'r') as f:
-            for line in f:
-                u, r, ha1 = line.rstrip().split(':')
-                if u == username and r == realm:
-                    result = ha1
-                    break
+        realm = kwargs.get('realm')
+        username = kwargs.get('username')
+        if realm and username:
+            with open(filename, 'r') as f:
+                for line in f:
+                    u, r, ha1 = line.rstrip().split(':')
+                    if u == username and r == realm:
+                        result = ha1
+                        break
         return result
 
     return get_ha1
 
 
-def synthesize_nonce(s, key, timestamp=None):
+def synthesize_nonce(s, key, algorithm, timestamp=None):
```

Comment:
Normally, we shouldn't change public API signatures in a way that reorders the args as this would be backwards-incompatible. Append new args to the end.